src/nodes.coffee

```diff
@@ -880,7 +880,19 @@ exports.Call = class Call extends Base
     content?.base.csx = yes
     fragments = [@makeCode('<')]
     fragments.push (tag = @variable.compileToFragments(o, LEVEL_ACCESS))...
-    fragments.push attributes.compileToFragments(o, LEVEL_PAREN)...
+    if attributes.base instanceof Arr
+      for obj in attributes.base.objects
+        attr = obj.base
+        attrProps = attr?.properties or []
+        # Catch invalid CSX attributes: <div {a:"b", props} {props} "value" />
+        if not (attr instanceof Obj or attr instanceof IdentifierLiteral) or (attr instanceof Obj and not attr.generated and (attrProps.length > 1 or not (attrProps[0] instanceof Splat)))
+          obj.error """
+            Unexpected token. Allowed CSX attributes are: id="val", src={source}, {props...} or attribute.
+            Example: <div id="val" src={getTheSource()} {props...} checked>hello world</div>.
```

Comment:
We don’t normally put examples in error messages. Please update the docs with this if the docs are insufficient.